src/Microsoft.AspNet.JsonPatch/Adapters/SimpleObjectAdapter.cs

```diff
@@ -167,40 +181,42 @@ private void Add(string path, object value, T objectToApplyTo, Operation<T> oper
                         {
                             throw new JsonPatchException<T>(operationToReport,
                                 string.Format("Patch failed: provided path is invalid for array property type at " +
-                                    "location path: {0}: position doesn't exist in array",
+                                    "location path: {0}: position larger than array size",
                                     path),
                                 objectToApplyTo);
                         }
                     }
-
-
                 }
                 else
                 {
                     throw new JsonPatchException<T>(operationToReport,
-                       string.Format("Patch failed: provided path is invalid for array property type at location path: {0}: expected array",
-                       path),
+                       string.Format("Patch failed: provided path is invalid for array property type at location " +
+                            "path: {0}: expected array",
+                            path),
                        objectToApplyTo);
                 }
             }
             else
             {
-                var conversionResultTuple = PropertyHelpers.ConvertToActualType(pathProperty.PropertyType, value);
+                var conversionResultTuple = PropertyHelpers.ConvertToActualType(
```

Comment:
I looked at this but could not figure out a direct replacement. Will sync up offline.